### PR TITLE
Add support for password in private key in paramiko { cli, variable }

### PR DESCRIPTION
**Pull request by fsbano**
Added support for password in private key on cli, variable, ansible.cfg

##### SUMMARY

Added support for password in private key in paramiko { cli, variable }

##### ISSUE TYPE

Added support for password in private key in paramiko { cli, variable } 

- Feature Pull Request

##### COMPONENT NAME

`lib/ansible/plugins/connection/paramiko_ssh.py`

##### ADDITIONAL INFORMATION

Added support for password in private key in paramiko { cli, variable }

### CODE REVIEW

**Comment by webknjaz** [comment]
> ##### COMPONENT NAME
> 
> Ansible - Core

Hey, this entire repository only contains core. I've updated your entry to be more accurate.

Also, such a change would definitely require tests to be included.

File: lib/ansible/plugins/connection/paramiko_ssh.py [hunk 1]
```diff
@@ -509,12 +531,16 @@ def _connect_uncached(self) -> paramiko.SSHClient:
             if LooseVersion(paramiko.__version__) >= LooseVersion('1.15.0'):
                 ssh_connect_kwargs['banner_timeout'] = self.get_option('banner_timeout')
 
+            passphrase = None
+            if self.get_option('private_key_file_passphrase'):
+                passphrase = self.get_option('private_key_file_passphrase')
```
Comment (webknjaz) [inline lib/ansible/plugins/connection/paramiko_ssh.py]: The if-block is redundant. You're essentially doing this one-liner:
```suggestion
            passphrase = self.get_option('private_key_file_passphrase') or None
```
(also, two calls to `get_option()` is probably expensive)

**Comment by fsbano** [comment]
> > ##### COMPONENT NAME
> 
> > 
> 
> > Ansible - Core
> 
> 
> 
> Hey, this entire repository only contains core. I've updated your entry to be more accurate.
> 
> 
> 
> Also, such a change would definitely require tests to be included.

Thank you very much !

Have nice day

Best Regards,
Fabio Sbano

**Comment by bcoca** [comment]
This is not really needed, if you provide the passphrase as the connection password, paramiko will automatically use it to decrypt the key

File: lib/ansible/config/base.yml [hunk 2]
```diff
@@ -930,6 +930,15 @@ DEFAULT_PRIVATE_KEY_FILE:
...
   type: path
+DEFAULT_PRIVATE_KEY_FILE_PASSPHRASE:
+  name: Private key file passphrase
+  default: ~
+  description:
+    - Private key file passphrase
+  env: [{name: ANSIBLE_PRIVATE_KEY_FILE_PASSPHRASE}]
+  ini:
+  - {key: private_key_file_passphrase, section: defaults}
+  type: string
```
Comment (bcoca) [inline lib/ansible/config/base.yml:941]: this should not be a global config if it only applies to a specific connection plugin

File: lib/ansible/plugins/connection/paramiko_ssh.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/plugins/connection/paramiko_ssh.py]: not really, as it should cache the option after the first access

**Comment by fsbano** [comment]
> This is not really needed, if you provide the passphrase as the connection password, paramiko will automatically use it to decrypt the key

Good Afternoon,

you say putting --ask-pass ? I confess that I have not tested it like this but anyway it will ask for the prompt

i want to run this by automation

**Comment by bcoca** [comment]
see issue #31988

**Comment by bcoca** [comment]
I was going to update the plugin and document this .. but past me beat me to it:
```
password:
   description:
      - Secret used to either login the ssh server or as a passphrase for ssh keys that require it
```

**Comment by fsbano** [comment]
> I was going to update the plugin and document this .. but past me beat me to it:
> 
> ```
> 
> password:
> 
>    description:
> 
>       - Secret used to either login the ssh server or as a passphrase for ssh keys that require it
> 
> ```

Very good !! 
I will go try again more late.. If it for true i will go cancel PR. 😃 

Best Regards,
Fabio Sbano

**Comment by fsbano** [comment]
> > I was going to update the plugin and document this .. but past me beat me to it:
> > ```
> > 
> > password:
> > 
> >    description:
> > 
> >       - Secret used to either login the ssh server or as a passphrase for ssh keys that require it
> > ```
> 
> Very good !! I will go try again more late.. If it for true i will go cancel PR. 😃
> 
> Best Regards, Fabio Sbano

@bcoca ,

Yes ! it worked perfectly --ask-pass opened the encrypted private key and I can also use the variable for silent mode

Best Regards,
Fabio Sbano

**Comment by fsbano** [comment]
fsbano@argon:~$ ansible-playbook -c paramiko -i ansible-hosts playbook.yml
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel" if you are modifying the Ansible engine, or trying out
features under development. This is a rapidly changing source of code and can become unstable at any point.

PLAY [argon] **********************************************************************************************************************************************************

TASK [Gathering Facts] ************************************************************************************************************************************************
[WARNING]: Platform linux on host 192.168.1.202 is using the discovered Python interpreter at /usr/bin/python3.11, but future installation of another Python
interpreter could change the meaning of that path. See https://docs.ansible.com/ansible-core/devel/reference_appendices/interpreter_discovery.html for more
information.
ok: [192.168.1.202]

TASK [Install latest version of Apache] *******************************************************************************************************************************
ok: [192.168.1.202]

PLAY RECAP ************************************************************************************************************************************************************
192.168.1.202              : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

fsbano@argon:~$

**Pull request closed without merging**